apps/balancer-tools/src/app/apr/round/(components)/TopPoolsChart.tsx

```diff
@@ -34,18 +46,14 @@ export default async function TopPoolsChart({ roundId }: { roundId: string }) {
     }
 
     const gauge = votingGauges[index];
-    const { apr } = await calculatePoolStats({
-      roundId,
-      poolId: gauge.pool.id,
-    });
+    const { apr } = await throttledCalculatePoolStats(roundId, gauge.pool.id);
```

Comment:
this is still iterating on a list and blocking rendering until we have every data we need, the only difference is that the requests are throttled. What value is this adding? Have you reflected about why this was needed here?

Reply:
The problem we're facing is that we're querying a bunch of pools. A direct implementation using Promise.all isn't suitable, as it would spam the server with simultaneous requests, resulting in immediate rate limiting. Consequently, these requests would persistently attempt retries, leading to subsequent requests slowing down. By using p-throttle, our intention is to navigate around the API's rate limitations. This approach should ideally enable us to accelerate the requests responses.